crates/ruff/src/rules/refurb/rules/unnecessary_enumerate.rs

```diff
@@ -0,0 +1,311 @@
+/// more efficient and communicates the intent of the code more clearly.
+///
+/// ## Example
+/// ```python
+/// for index, value in enumerate(sequence):
+///     print(index)
+///
+/// for index, value in enumerate(sequence):
+///     print(value)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// for index in range(len(sequence)):
+///     print(index)
+///
+/// for value in sequence:
+///     print(value)
+/// ```
+///
+/// ## References
+/// - [Python documentation: `enumerate`](https://docs.python.org/3/library/functions.html#enumerate)
+#[violation]
+pub struct UnnecessaryEnumerate {
+    subset: EnumerateSubset,
+    iterable_suggestion: Option<SourceCodeSnippet>,
+}
+
+impl Violation for UnnecessaryEnumerate {
+    const AUTOFIX: AutofixKind = AutofixKind::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let UnnecessaryEnumerate {
+            subset,
+            iterable_suggestion,
+        } = self;
+        if let Some(suggestion) = iterable_suggestion
+            .as_ref()
+            .and_then(SourceCodeSnippet::full_display)
+        {
+            format!("Do not iterate over `enumerate` when only using the {subset}, iterate over `{suggestion}` instead")
+        } else {
+            format!("Do not iterate over `enumerate` when only using the {subset}")
+        }
+    }
+
+    fn autofix_title(&self) -> Option<String> {
+        let UnnecessaryEnumerate {
+            subset,
+            iterable_suggestion,
+        } = self;
+        if let Some(suggestion) = iterable_suggestion
+            .as_ref()
+            .and_then(SourceCodeSnippet::full_display)
+        {
+            Some(format!("Replace with `{suggestion}`"))
+        } else {
+            match subset {
+                EnumerateSubset::Indices => {
+                    Some("Replace with iteration over `range(len(...))`".to_string())
+                }
+                EnumerateSubset::Values => {
+                    Some("Replace with direct iteration over the sequence".to_string())
+                }
+            }
+        }
+    }
+}
+
+/// FURB148
+pub(crate) fn unnecessary_enumerate(checker: &mut Checker, stmt_for: &ast::StmtFor) {
+    // Check the for statement is of the form `for x, y in func(...)`.
+    let Expr::Tuple(ast::ExprTuple { elts, .. }) = stmt_for.target.as_ref() else {
+        return;
+    };
+    let [index, value] = elts.as_slice() else {
+        return;
+    };
+    let Expr::Call(ast::ExprCall {
+        func, arguments, ..
+    }) = stmt_for.iter.as_ref()
+    else {
+        return;
+    };
+
+    // Check that the function is the `enumerate` builtin.
+    let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() else {
+        return;
+    };
+    if id != "enumerate" {
+        return;
+    };
+    if !checker.semantic().is_builtin("enumerate") {
+        return;
+    };
+
+    // Get the `start` argument, if it is a constant integer.
+    let start = start(arguments);
```

Comment:
nit: maybe we could move this to the respective match branch where it's being used.